tests/conftest.py

```diff
@@ -779,35 +773,48 @@ def generate_priority_lists(request, prio_scope):
 
     return ret if ret else empty
 
+
+_duthosts_in_testbed = None
+_duts_vars_in_inv = {}
 def pytest_generate_tests(metafunc):
     # The topology always has atleast 1 dut
     dut_indices = [0]
-    tbname, testbedinfo = get_tbinfo(metafunc)
-    duts_in_testbed = testbedinfo["duts"]
+    global _duthosts_in_testbed, _frontend_hosts_per_hwsku_per_module
+    tbname = None
+    if _duthosts_in_testbed is None:
+        tbname, testbedinfo = get_tbinfo(metafunc)
+        _duthosts_in_testbed = testbedinfo["duts"]
+        # Get vars defined for each DUT in the inventory
+        inv_files = get_inventory_files(metafunc)
+        for dutname in _duthosts_in_testbed:
+            _duts_vars_in_inv[dutname] = get_host_visible_vars(inv_files, dutname)
```

Comment:
This change is still causing all testcases to try collect testbed_info once, so if a test doesn't need any enumeration, it will still collect the testbed info once?

Would it make sense to put the collector and global variable cache in a helper function and call when needed?